doc/man7/pbkdf2.pod

```diff
@@ -0,0 +1,129 @@
+=pod
+
+=head1 NAME
+
+pbkdf2 - PBKDF2 key derivation function
+
+=head1 SYNOPSIS
+
+ #include <openssl/kdf.h>
+
+ int EVP_PKEY_CTX_set1_pbe_pass(EVP_PKEY_CTX *pctx, unsigned char *pass,
+                                int passlen);
+
+ int EVP_PKEY_CTX_set1_pbkdf2_salt(EVP_PKEY_CTX *pctx, unsigned char *salt,
+                                   int saltlen);
+
+ int EVP_PKEY_CTX_set_pbkdf2_md(EVP_PKEY_CTX *pctx, const EVP_MD *md);
+
+ int EVP_PKEY_CTX_set_pbkdf2_iter(EVP_PKEY_CTX *pctx, int itercnt);
+
+=head1 DESCRIPTION
+
+The EVP_PKEY_PBKDF2 algorithm implements the PBKDF2 password based key
+derivation function, as described in RFC 2898. It is an iterative HMAC-based
+key derivation function that offers scaling by a linear work factor. PBKDF2 is
+not memory-hard.
```

Comment:
"Stronger Key Derivation via Sequential Memory-Hard Functions", Percival 2009:

> Definition 1. A memory-hard algorithm on a Random Access Machine is an al-
> gorithm which uses S(n) space and T (n) operations, where S(n) ∈ Ω T (n) 1−ǫ .
> 
> A memory-hard algorithm is thus an algorithm which asymptotically uses almost
> as many memory locations as it uses operations 5 ; it can also be thought of as an
> algorithm which comes close to using the most memory possible for a given number
> of operations, since by treating memory addresses as keys to a hash table it is trivial
> to limit a Random Access Machine to an address space proportional to its running
> time.